File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.17.2.c
/* { dg-do compile } */

void
a17_2_wrong ()
{
  int x;
  int *i;
  float *r;
  i = &x;
  r = (float *) &x;
#pragma omp parallel
  {
#pragma omp atomic
    *i += 1;
#pragma omp atomic
    *r += 1.0;
/* Incorrect because the atomic constructs reference the same location
     through incompatible types */
  }
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.17.2.f90
! { dg-do compile }

      SUBROUTINE SUB()
        COMMON /BLK/ R
        REAL R
!$OMP ATOMIC
        R = R + 1.0
      END SUBROUTINE SUB

      SUBROUTINE A17_2_WRONG()
      COMMON /BLK/ I
      INTEGER I
!$OMP PARALLEL
!$OMP ATOMIC
            I=I+1
          CALL SUB()
!$OMP END PARALLEL
      END SUBROUTINE A17_2_WRONG

